test.functional/config/_templates/template-puppeteer-environment.js

```diff
@@ -0,0 +1,14 @@
+const PuppeteerEnvironment = require('../config/common/puppeteer-common-environment.js');
+
+class Env extends PuppeteerEnvironment {
+    constructor(config) {
+        super(config);
+    }
+
+    async setup() {
+        await super.setup();
+        this.global.httpUrl = '{{instanceUrl}}';
```

Comment:
`{{instance.json}}`

```
interface Instance {
@get:Internal
@get:JsonIgnore
val json: String
    get() = Formats.toJson(this)
}
```

or as method `toJson()` to avoid annotations

Reply:
Done